### ISSUE TITLE
LdCalculator pairwise differences

### ISSUE THREAD

**Issue by tcavazos2** [tskit-dev/tskit#396]
When using the `LdCalculator.get_r2_array()` there seems to be large pairwise differences between SNPs. The LD between snp1 and snp2 is different than the LD between snp2 and snp1. This may be an error in one direction (`msprime.FORWARD` or `msprime.REVERSE`). See below example code detailing this issue:

``` python

import msprime

def get_ld(mutations):
    results = {}
    for focal_mutation in mutations:
        a = ld_calc.get_r2_array(
        focal_mutation, max_distance=1e6,
        direction=msprime.REVERSE)
        rev_indexes = focal_mutation - np.nonzero(a >= 0)[0] - 1
        b = ld_calc.get_r2_array(
            focal_mutation, max_distance=1e6,
            direction=msprime.FORWARD)
        fwd_indexes = focal_mutation + np.nonzero(b >= 0)[0] + 1
        indexes = np.concatenate((rev_indexes[::-1], fwd_indexes))
        r2_vals = list(a[::-1])+list(b)
        results[focal_mutation] = dict(zip(indexes,r2_vals))
    return results

ts = msprime.simulate(100, recombination_rate=10, mutation_rate=20, random_seed=1)
ld_calc = msprime.LdCalculator(ts)
results = get_ld([1,2])

ld_1_2, ld_2_1 = results.get(1).get(2), results.get(2).get(1) 

print("LD between variant 1 and variant 2 = {}".format(ld_1_2))
print("LD between variant 2 and variant 1 = {}".format(ld_2_1))
print("Absolute difference = {}".format(abs(ld_1_2-ld_2_1)))

```

```
> LD between variant 1 and variant 2 = 0.6767676767676769
> LD between variant 2 and variant 1 = 0.02050811141720233
> Absolute difference = 0.6562595653504746
```

**Comment by tcavazos2** [comment]
This may be a bug specific to `get_r2_array()` because the below code yields the same value in both directions:

```python
print(ld_calc.get_r2(1,2))
print(ld_calc.get_r2(2,1))
```
```
> 0.6767676767676769
> 0.6767676767676769
```

**Comment by jeromekelleher** [comment]
Thanks very much for the bug report @tcavazos2. It turns out that this is a subtle issue with the way your code is laid out, but I think should be regarded as a bug in the library (I've opened a PR to fix it in #397) because it's such a nasty gotcha.

Basically, the underlying memory for your ``a`` and ``b`` arrays is the same, and at the moment, you need to make a copy of ``a`` before you call ``get_r2_array`` again, or it'll be overwritten. So, if you change it to:
```python
def get_ld(mutations):
    results = {}
    for focal_mutation in mutations:
        a = ld_calc.get_r2_array(
            focal_mutation, max_distance=1e6,
            direction=msprime.REVERSE)
        # Take a copy of a
        a = a.copy()
        # Same as before, from here.  
```

Apologies for this gotcha, and thanks again for reporting it.

**Comment by tcavazos2** [comment]
This makes sense! Thanks so much.
